core/src/services/routing.js

```diff
@@ -56,22 +56,17 @@ class RoutingClass {
     if (windowPath === GenericHelpers.trimLeadingSlash(route)) {
       return;
     }
+    const hashRouting = LuigiConfig.getConfigValue('routing.useHashRouting');
+    let url = new URL(location.href);
+    hashRouting ? url.hash = route : url.pathname = route;
 
-    if (LuigiConfig.getConfigValue('routing.useHashRouting')) {
-      if (!navSync) {
-        EventListenerHelpers.hashChangeWithoutSync = true;
-      }
-      window.location.hash = route;
-      return;
-    }
-
-    const method = pushState ? 'pushState' : 'replaceState';
+    const method = LuigiConfig.getConfigValue('routing.disableBrowserHistory') ? 'replaceState' : pushState ? 'pushState' : 'replaceState';
```

Comment:
I feel this line is not easy to understand. Could you please briefly explain the logic?

Does the logic same as the code below?
```
const method = LuigiConfig.getConfigValue('routing.disableBrowserHistory') && pushState ? 'pushState' : 'replaceState';
```

Reply:
If I remember right I think it goes like `const method = A && B ? C : D`
step 1 - check if A is true, if it is true return C - **end**
step 2 - A is false, then check B, if B is true return C, else return D -  **end**
but yeah this could be maybe broken up into pieces so it can be easier debugged I guess

Reply:
I could see this as more self-explanatory: 
```
const chosenHistoryMethod = pushState ? 'pushState' : 'replaceState';
const method = LuigiConfig.getConfigValue('routing.disableBrowserHistory') ? 'replaceState' : chosenHistoryMethod;
```
This would maybe not work in case **disableBrowserHistory** is not used and `pushState=true` by default, it would return **replaceState** instead of intended **pushState**
```
const method = LuigiConfig.getConfigValue('routing.disableBrowserHistory') && pushState ? 'pushState' : 'replaceState';
```
Maybe this would work as one-liner 🤔 ?
```
const method = LuigiConfig.getConfigValue('routing.disableBrowserHistory') || !pushState ? 'replaceState' : 'pushState'  ;
```

Reply:
Yeah, not necessary to be one line if two lines of code are clearer.

Reply:
Go with the two-liner